Trie/Trie/Trie.cs

```diff
@@ -0,0 +1,214 @@
+
+        if (nextPosition == element.Length)
+        {
+            if (!endPrefixNode.IsTerminal)
+            {
+                endPrefixNode.IsTerminal = true;
+                return true;
+            }
+
+            return false;
+        }
+
+        endPrefixNode.Children[element[nextPosition]] = CreateSuffix(element.Substring(nextPosition));
+        return true;
+    }
+
+    /// <summary>
+    /// Removes a string from the Trie.
+    /// </summary>
+    /// <param name="element">The string to remove from the Trie.</param>
+    /// <returns>
+    /// <c>true</c> if the string was successfully removed;
+    /// <c>false</c> if the string wasn't found in the Trie.
+    /// </returns>
+    /// <exception cref="ArgumentException">
+    /// Thrown when the input string is null or empty.
+    /// </exception>
+    public bool Remove(string element)
+    {
+        ArgumentNullException.ThrowIfNull(element, nameof(element));
+        if (element.Length == 0)
+        {
+            throw new ArgumentException("String cannot be empty", nameof(element));
+        }
+
+        var (endPrefixNode, nextPosition, pathStack) = this.FindLongestPrefix(element, true);
+
+        if (nextPosition != element.Length || endPrefixNode.IsTerminal == false)
+        {
+            return false;
+        }
+
+        char previousSymbol = '\0';
+        bool isPreviousLeaf = false;
+        endPrefixNode.IsTerminal = false;
+
+        while (pathStack!.Count > 0)
+        {
+            Node last = pathStack.Pop();
+
+            if (isPreviousLeaf)
+            {
+                last.Children.Remove(previousSymbol);
+                isPreviousLeaf = false;
+            }
+
+            if (last.Children.Count == 0 && !last.IsTerminal)
+            {
+                isPreviousLeaf = true;
+                previousSymbol = last.Value;
+            }
+            else
+            {
+                return true;
+            }
+        }
+
+        if (isPreviousLeaf)
+        {
+            this.Root.Children.Remove(previousSymbol);
+        }
+
+        return true;
+    }
+
+    /// <summary>
+    /// Determines whether the Trie contains the specified string.
+    /// </summary>
+    /// <param name="element">The string to locate in the Trie.</param>
+    /// <returns>
+    /// <c>true</c> if the Trie contains the string;
+    /// <c>false</c> otherwise.
+    /// </returns>
+    /// <exception cref="ArgumentException">
+    /// Thrown when the input string is null or empty.
+    /// </exception>
+    public bool Contains(string element)
+    {
+        ArgumentNullException.ThrowIfNull(element, nameof(element));
+        if (element.Length == 0)
+        {
+            throw new ArgumentException("String cannot be empty", nameof(element));
+        }
+
+        var (endPrefixNode, nextPosition, pathStack) = this.FindLongestPrefix(element, false);
+
+        return nextPosition == element.Length && endPrefixNode.IsTerminal == true;
```

Comment:
```suggestion
        return nextPosition == element.Length && endPrefixNode.IsTerminal;
```
Мне кажется, сравниваться с булевыми константами ещё в первом семестре строжайше запретили :)